src/main/java/com/gary/backendv2/service/TrustedPersonService.java

```diff
@@ -0,0 +1,114 @@
+package com.gary.backendv2.service;
+
+import com.gary.backendv2.exception.HttpException;
+import com.gary.backendv2.model.TrustedPerson;
+import com.gary.backendv2.model.User;
+import com.gary.backendv2.model.dto.request.AllergyRequest;
+import com.gary.backendv2.model.dto.request.TrustedPersonRequest;
+import com.gary.backendv2.model.dto.response.TrustedPersonResponse;
+import com.gary.backendv2.repository.TrustedPersonRepository;
+import com.gary.backendv2.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class TrustedPersonService {
+	private final TrustedPersonRepository trustedPersonRepository;
+	private final UserRepository userRepository;
+
+	public List<TrustedPersonResponse> getAll(){
+		List<TrustedPerson> trustedPeople = trustedPersonRepository.findAll();
+		List<TrustedPersonResponse> trustedPersonResponses = new ArrayList<>();
+		for (TrustedPerson t: trustedPeople) {
+			trustedPersonResponses.add(
+				TrustedPersonResponse
+						.builder()
+						.email(t.getEmail())
+						.firstName(t.getFirstName())
+						.trustedId(t.getTrustedId())
+						.lastName(t.getLastName())
+						.phone(t.getPhone())
+						.build()
+			);
+		}
+		return trustedPersonResponses;
+	}
+
+	public TrustedPersonResponse getByEmail(String userEmail){
+		Optional<User> userOptional = userRepository.findByEmail(userEmail);
+		if (userOptional.isEmpty()) {
+			throw new HttpException(HttpStatus.NOT_FOUND, String.format("Cannot find user with %s", userEmail));
+		}
+		TrustedPerson trustedPerson = userOptional.get().getTrustedPerson();
+		if(trustedPerson == null){
+			throw new HttpException(HttpStatus.NO_CONTENT, String.format("User %s don't have trusted person", userEmail));
+		}
```

Comment:
The HTTP 204 No Content success status response code indicates that a request has succeeded, but that the client doesn't need to navigate away from its current page.

404 Not Found pls

Reply:
ewentualnie 200 OK ale bez response body